### PR TITLE
Fix TM test location

### PR DESCRIPTION
**Pull request by shreyas-s-rao**
Signed-off-by: Shreyas Rao <shreyas.sriganesh.rao@sap.com>

**What this PR does / why we need it**:
This PR makes sure that the correct version of the code is pulled for TM tests on PRs. Also fixes TEST_ID generation for TM tests.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

**Review by abdasgupta — APPROVED** [review]
/lgtm
Should we merge this now?

File: .ci/tm-tests-playground [hunk 1]
```diff
@@ -84,6 +84,9 @@ echo "Testmachinery landscape: ${TM_LANDSCAPE}"
 echo "Arguments: ${ARGUMENTS}"
 echo "EFFECTIVE_VERSION: ${EFFECTIVE_VERSION}"
 
+GIT_REVISION="${EFFECTIVE_VERSION##*\-}"
```
Comment (amshuman-kr) [inline .ci/tm-tests-playground:87]: Could you please explain what this is about?

**Review by amshuman-kr — APPROVED** [review]
Thanks @shreyas-s-rao! /lgtm

Just one question below which is for my own curiosity :-)

**Comment by abdasgupta** [comment]
The tm tests are failing here. https://grafana.ingress.tm.core.shoot.live.k8s-hana.ondemand.com/explore?orgId=1&left=%5B%22now-3d%22,%22now%22,%22Loki%22,%7B%22expr%22:%22%7Bcontainer%3D%5C%22main%5C%22,instance%3D%5C%22playground5jx46-wf-1003531664%5C%22%7D%22%7D,%7B%22mode%22:%22Logs%22%7D%5D

File: .ci/tm-tests-playground (hunk 1 above)
Comment (shreyas-s-rao) [reply .ci/tm-tests-playground:87]: This operator splits the EFFECTIVE_VERSION string which looks like `v0.13.0-dev-<git_commit_hash>` and gets the last element.

**Pull request merged**